composer/loggers/neptune_logger.py

```diff
@@ -304,4 +324,37 @@ def _validate_image(img: Union[np.ndarray, torch.Tensor], channels_last: bool) -
     if not channels_last:
         img_numpy = np.moveaxis(img_numpy, 0, -1)
 
-    return img_numpy
+    return _validate_image_value_range(img_numpy)
+
+
+def _validate_image_value_range(img: np.ndarray) -> np.ndarray:
+    array_min = img.min()
+    array_max = img.max()
+
+    if (array_min >= 0 and 1 < array_max <= 255) or (array_min >= 0 and array_max <= 1):
+        return img
+
+    from neptune.common.warnings import NeptuneWarning, warn_once
+
+    warn_once(
+        'Image value range is not in the expected range of [0.0, 1.0] or [0, 255]. '
+        'This might be due to the presence of `transforms.Normalize` in the data pipeline. '
+        'Logged images may not display correctly in Neptune.',
+        exception=NeptuneWarning,
+    )
+
+    return _scale_image_to_0_255(img, array_min, array_max)
```

Comment:
If we are already warning that the images might not be displayed correctly, let's not scale them anymore

Reply:
IMO, that's acceptable. (ideally, we should only `warn_once` unless we can tell exactly which image triggered a warning, but that's a discussion on the client side)

Being warned that the image might not be displayed correctly and then seeing that we've distorted images further by scaling them might be more confusing to the users.

Reply:
They will anyway be distorted further, because we need to send a png

Reply:
And if we don't scale here, we will get sth like
```
[neptune] [warning] The smallest value in the array is -1.3407506942749023 and the largest value in the array is 1.7876918315887451. To be interpreted as colors correctly values in the array need to be in the [0, 1] range.
[neptune] [warning] The smallest value in the array is -1.3407506942749023 and the largest value in the array is 1.7411906719207764. To be interpreted as colors correctly values in the array need to be in the [0, 1] range.
[neptune] [warning] The smallest value in the array is -1.8401265144348145 and the largest value in the array is 1.968528151512146. To be interpreted as colors correctly values in the array need to be in the [0, 1] range.
[neptune] [warning] The smallest value in the array is -1.6932512521743774 and the largest value in the array is 1.599104881286621. To be interpreted as colors correctly values in the array need to be in the [0, 1] range.
[neptune] [warning] The smallest value in the array is -1.9023436307907104 and the largest value in the array is 2.025362491607666. To be interpreted as colors correctly values in the array need to be in the [0, 1] range.
[neptune] [warning] The smallest value in the array is -1.752001404762268 and the largest value in the array is 1.911994457244873. To be interpreted as colors correctly values in the array need to be in the [0, 1] range.
[neptune] [warning] The smallest value in the array is -1.9023436307907104 and the largest value in the array is 1.7269821166992188. To be interpreted as colors correctly values in the array need to be in the [0, 1] range.
[neptune] [warning] The smallest value in the array is -1.3701257705688477 and the largest value in the array is 2.025362491607666. To be interpreted as colors correctly values in the array need to be in the [0, 1] range.
```

Reply:
Yeah. This is what I was referring to with a `warn_once`, because multiple such warnings without providing which image triggered the warning are not helpful.

We can go ahead with scaling if the images are going to be distorted for rendering on the app anyway